### PR TITLE
Add list plugin CLI command

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: src/Console/Plugin/ListCommand.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $plug     = new Plugin();
+        $pluglist = $plug->find([], "name, directory");
+        $data = [];
+        foreach ($pluglist as $plugin) {
+            $record = [];
+            $record['name'] = Toolbox::stripTags($plugin['name']);
+            $record['version'] = Toolbox::stripTags($plugin['version']);
```
Comment (cedric-anne) [inline src/Console/Plugin/ListCommand.php:76]: On CLI context, I think there is no reason to protect output from HTML injection.

```suggestion
            $record['name'] = $plugin['name'];
            $record['version'] = $plugin['version'];
```

File: src/Console/Plugin/ListCommand.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            $record['install_method'] = file_exists(GLPI_MARKETPLACE_DIR . "/" . $plugin['directory']) ? Controller::getTypeName() : __("Manually installed");
+            $data[] = $record;
+        }
+        $format = $input->getOption('format');
+
+        if ($format === 'json') {
+            $output->writeln(json_encode($data));
+        } else {
+            $table = new Table($output);
+            $table->setHeaders(['Name', 'Version', 'State', 'Install method']);
```
Comment (cedric-anne) [inline src/Console/Plugin/ListCommand.php:87]:
```suggestion
            $table->setHeaders([__('Name'), __('Version'), __('Status'), __('Install method')]);
```

**Review by cedric-anne — COMMENTED** [review]
It can be usefull to have either the plugin `key` (e.g. `stab`), either the plugin path on filesystem (e.g. `/var/www/glpi/plugins/stab`).

**Comment by cconard96** [comment]
All changes from suggestions have been made. The plugin key is now shown in the output. I added an option flag to display the full plugin path as well (not shown by default).

**Review by cedric-anne — APPROVED** [review]

**Comment by cconard96** [comment]
One last commit to ensure the correct path separator for the OS is shown in case someone hosts GLPI on Windows.

File: src/Console/Plugin/ListCommand.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            }
+            $data[] = $record;
+        }
+        $format = $input->getOption('format');
+
+        if ($format === 'json') {
+            $output->writeln(json_encode($data));
+        } else {
+            $table = new Table($output);
+            $headers = [__('Plugin Key'), __('Name'), __('Version'), __('Status'), __('Install method')];
```
Comment (trasher) [inline src/Console/Plugin/ListCommand.php:101]:
```suggestion
            $headers = [__('Plugin Key'), __('Name'), _n('Version', 'Versions', 1), __('Status'), __('Install method')];
```

**Pull request merged**